LayoutKit/Layouts/TextViewLayout.swift

```diff
@@ -0,0 +1,163 @@
+                            layoutAlignment: Alignment = defaultAlignment,
+                            flexibility: Flexibility = defaultFlexibility,
+                            viewReuseId: String? = nil,
+                            configure: ((TextView) -> Void)? = nil) {
+        self.init(text: .attributed(attributedText),
+                  font: font,
+                  lineFragmentPadding: lineFragmentPadding,
+                  contentInset: contentInset,
+                  layoutMargins: layoutMargins,
+                  textContainerInset: textContainerInset,
+                  layoutAlignment: layoutAlignment,
+                  flexibility: flexibility,
+                  viewReuseId: viewReuseId,
+                  configure: configure)
+    }
+
+    // MARK: - Layout protocol
+
+    open func measurement(within maxSize: CGSize) -> LayoutMeasurement {
+        let fittedSize = textSize(within: maxSize)
+        return LayoutMeasurement(layout: self, size: fittedSize.decreasedToSize(maxSize), maxSize: maxSize, sublayouts: [])
+    }
+
+    open func arrangement(within rect: CGRect, measurement: LayoutMeasurement) -> LayoutArrangement {
+        let frame = alignment.position(size: measurement.size, in: rect)
+        return LayoutArrangement(layout: self, frame: frame, sublayouts: [])
+    }
+
+    // MARK: - private helpers
+
+    private func textSize(within maxSize: CGSize) -> CGSize {
+        var size = TextCalculatorHelper.textSize(maxSize: maxSize, text: text, font: font)
+
+        let heightInset = (textContainerInset.top + textContainerInset.bottom) + (layoutMargins.top + layoutMargins.bottom)
+        let widthInset = (textContainerInset.left + textContainerInset.right) + (layoutMargins.left + layoutMargins.right) + lineFragmentPadding * 2
+            + (contentInset.left + contentInset.right)
+        size = CGSize(width: size.width + widthInset, height: size.height + heightInset)
+
+        return size
+    }
+
+    // MARK: - overriden methods
+
+    /// Don't change `textContainerInset`, `lineFragmentPadding`, `contentInset`
+    /// and `layoutMargins` in `configure`. By changing those, it will cause the incorrect
+    /// size calculation. So they will be reset by using parameters from initializer.
+    /// `usesFontLeading` doesn't support in `TextViewLayout`.
+    open override func configure(view textView: TextView) {
+        config?(textView)
+        textView.textContainerInset = textContainerInset
+        textView.textContainer.lineFragmentPadding = lineFragmentPadding
+        textView.contentInset = contentInset
+        textView.layoutMargins = layoutMargins
+        textView.layoutManager.usesFontLeading = false
+        textView.font = font
+        switch text {
+        case .unattributed(let text):
+            textView.text = text
+        case .attributed(let attributedText):
+            textView.attributedText = attributedText
+        }
+    }
+
+    open override var needsView: Bool {
```

Comment:
Right now, LabelLayout just returns true and doesn't try to be fancy about this. I am not quite sure what it right myself, but it seems like this should be consistent with that either way.